src/gui/owncloudgui.cpp

```diff
@@ -742,6 +744,64 @@ void ownCloudGui::setupActions()
     }
 }
 
+void ownCloudGui::fetchNavigationApps(AccountStatePtr account, QMenu *accountMenu){
+    OcsNavigationAppsJob *job = new OcsNavigationAppsJob(account->account());
+    job->setProperty(propertyAccountC, QVariant::fromValue(account->account()));
+    job->setProperty(propertyMenuC, QVariant::fromValue(accountMenu));
+    connect(job, &OcsNavigationAppsJob::appsJobFinished, this, &ownCloudGui::slotNavigationAppsFetched);
+    connect(job, &OcsNavigationAppsJob::ocsError, this, &ownCloudGui::slotOcsError);
+    job->getNavigationApps();
+}
+
+void ownCloudGui::slotNavigationAppsFetched(const QJsonDocument &reply)
+{
+    if(!reply.isEmpty()){
+        auto element = reply.object().value("ocs").toObject().value("data");
+        auto navLinks = element.toArray();
+        if(navLinks.size() > 0){
+            if(auto account = qvariant_cast<AccountPtr>(sender()->property(propertyAccountC))){
+                if(QMenu *accountMenu = qvariant_cast<QMenu*>(sender()->property(propertyMenuC))){
+
+                    // when there is only one account add the nav links above the settings
+                    QAction *actionBefore = _actionSettings;
+
+                    // when there is more than one account add the nav links above pause/unpause folder or logout action
+                    if(AccountManager::instance()->accounts().size() > 1){
+                        foreach(QAction *action, accountMenu->actions()){
+
+                            // pause/unpause folder and logout actions have propertyAccountC
+                            if(auto actionAccount = qvariant_cast<AccountStatePtr>(action->property(propertyAccountC))){
+                                if(actionAccount->account() == account){
+                                    actionBefore = action;
+                                    break;
+                                }
+                             }
+                        }
+                    }
+
+                    // Create submenu with links
+                    QMenu *navLinksMenu = new QMenu(tr("Apps"));
+                    accountMenu->insertSeparator(actionBefore);
+                    accountMenu->insertMenu(actionBefore, navLinksMenu);
+                    foreach (const QJsonValue &value, navLinks) {
+                        auto navLink = value.toObject();
+                        QAction *action = new QAction(navLink.value("name").toString(), this);
+                        QUrl href(account->url().host() + navLink.value("href").toString());
```

Comment:
You can just add absolute=true as a get parameter to the OCS request, so the href value already contains the full URL.

Reply:
Besides that opening fails for me with the following error, when clicking the menu item:

```gio: file:///home/jus/repos/nextcloud/build-client-Desktop-Debug/bin/localhost/index.php/apps/contacts: Could not fetch information from »/home/jus/repos/nextcloud/build-client-Desktop-Debug/bin/localhost/index.php/apps/contacts«: File for directory not found```

Reply:
Done. Please, try it again.